### PR TITLE
Change error code to 404 when account not exist

### PR DESCRIPTION
**Pull request by blueww**
Thanks for contribution! Please go through following checklist before sending PR.
To Fix https://github.com/Azure/Azurite/issues/1816

### PR Branch Destination

- For Azurite V3, please send PR to `main` branch.
- For legacy Azurite V2, please send PR to `legacy-dev` branch.

### Always Add Test Cases

Make sure test cases are added to cover the code change.

### Add Change Log

Add change log for the code change in `Upcoming Release` section in `ChangeLog.md`.

### Development Guideline

Please go to CONTRIBUTION.md for steps about setting up development environment and recommended Visual Studio Code extensions.

### CODE REVIEW

File: src/table/authentication/TableSharedKeyAuthenticator.ts [hunk 1]
```diff
@@ -43,9 +43,8 @@ export default class TableSharedKeyAuthenticator implements IAuthenticator {
         `TableSharedKeyAuthenticator:validate() Invalid storage account ${account}.`,
         tableContext.contextID
       );
-      throw StorageErrorFactory.getInvalidOperation(
-        tableContext,
-        "Invalid storage account."
+      throw StorageErrorFactory.ResourceNotFound(
+        context
```
Comment (EmmaZhu) [inline src/table/authentication/TableSharedKeyAuthenticator.ts:47]: What's the difference between this context and tableContext?

File: tests/blob/apis/container.test.ts [hunk 2]
```diff
@@ -1211,4 +1211,34 @@ describe("ContainerAPIs", () => {
       await blob.delete();
     }
   });
+
+  it("Create container with not exist Account, return 404 @loki @sql", async () => {
```
Comment (EmmaZhu) [inline tests/blob/apis/container.test.ts:1215]: Have you tested with other credentials and services?

File: src/table/authentication/TableSharedKeyAuthenticator.ts (hunk 1 above)
Comment (blueww) [reply src/table/authentication/TableSharedKeyAuthenticator.ts:47]: context and tableContext are in differenct type, and tableContext's type is inherit from context 's type. 
And tableContext is created from context. 
`    const tableContext = new TableStorageContext(context);`

Here, both should work, as the input should be in type of Context.

File: tests/blob/apis/container.test.ts (hunk 2 above)
Comment (blueww) [reply tests/blob/apis/container.test.ts]: Add test cases to cover all service * all credentials

**Review by sreekara123 — APPROVED** [review]

**Review by XiaoningLiu — APPROVED** [review]

**Pull request merged**